src/Microsoft.AspNetCore.WebUtilities/HttpResponseStreamWriter.cs

```diff
@@ -186,22 +220,48 @@ public override async Task WriteAsync(char[] values, int index, int count)
                 }
 
                 CopyToCharBuffer(values, ref index, ref count);
+                Debug.Assert(count == 0);
             }
         }
 
-        public override async Task WriteAsync(string value)
+        public override Task WriteAsync(string value)
         {
             if (_disposed)
             {
-                throw new ObjectDisposedException(nameof(HttpResponseStreamWriter));
+                return GetObjectDisposedTask();
             }
 
             if (value == null)
             {
-                return;
+                return Task.CompletedTask;
+            }
+
+            var count = value.Length;
+            if (count == 0)
+            {
+                return Task.CompletedTask;
+            }
```

Comment:
Is it worth folding this together with L234?

Reply:
Done